System/Random.hs

```diff
@@ -131,8 +133,22 @@ class RandomGen g where
   -- |The 'next' operation returns an 'Int' that is uniformly distributed
   -- in the range returned by 'genRange' (including both end points),
   -- and a new generator.
-  next     :: g -> (Int, g)
-  -- `next` can be deprecated over time
+  {-# DEPRECATE next "Use genWord32[R] or genWord64[R]" #-}
+  next :: g -> (Int, g)
+  next g = (minR + fromIntegral w, g') where
+    (minR, maxR) = genRange g
+    range = fromIntegral $ maxR - minR
```

Comment:
This implementation will overflow

Reply:
Example:
```haskell
Prelude> maxR = maxBound ::Int
Prelude> maxR - (-1)
-9223372036854775808
```

Reply:
Although due modulo arithmetics it might work out in the end

Reply:
The type of `range` is `Word64` or `Word32`, depending on the size of `Int`.

Without a conversion, I also get an overflow:

```haskell
*System.Random> (maxBound :: Int) - (minBound :: Int)
-1
```

But the relevant case is where the type of `range` is known to be the word of the correct size:

```haskell
*System.Random> fromIntegral ((maxBound :: Int) - (minBound :: Int)) :: Word64
18446744073709551615
```

AFAICT it's the latter that happens on line 140.